src/main/java/io/vertx/core/impl/DeploymentManager.java

```diff
@@ -79,13 +99,47 @@ public void deployVerticle(Verticle verticle, DeploymentOptions options,
       throw new IllegalArgumentException("Can't specify isolatedClasses for already created verticle");
     }
     ContextImpl currentContext = vertx.getOrCreateContext();
-    doDeploy("java:" + verticle.getClass().getName(), generateDeploymentID(), options, currentContext, currentContext, completionHandler,
-        getCurrentClassLoader(), verticle);
+    ClassLoader cl = getClassLoader(options, currentContext);
+    currentContext.<Set<Verticle>>executeBlocking(fut -> {
```

Comment:
I think it should simply do nothing, most likely add a note that the supplier will be called on the current thread and so if the user is on an event loop he will know that he should not block the creation, if he has a blocking supplier he would himself wrap it with an executeBlocking block.